comet-java-client/src/test/java/ml/comet/experiment/impl/OnlineExperimentTest.java

```diff
@@ -445,22 +448,66 @@ public void testUploadAndGetAssets() {
     }
 
     @Test
-    public void testSetsContext() {
-        OnlineExperiment experiment = createOnlineExperiment();
+    public void testLogAndGetRemoteAssets() {
+        try (OnlineExperimentImpl experiment = (OnlineExperimentImpl) createOnlineExperiment()) {
+            // Make sure experiment has no assets
+            //
+            assertTrue(experiment.getAssetList(ASSET_TYPE_ALL).isEmpty());
+
+            // Log remote assets and wait for completion
+            //
+            ExperimentContext context = new ExperimentContext(10, 101, "train");
+            OnCompleteAction onComplete = new OnCompleteAction();
+            Map<String, Object> metadata = new HashMap<String, Object>() {{
+                put("someInt", 10);
+                put("someString", "test string");
+                put("someBoolean", true);
+            }};
+            String metadataJson = JsonUtils.toJson(metadata);
+
+            URI firstAssetLink = new URI("s3://bucket/folder/firstAssetFile.extension");
+            String firstAssetFileName = "firstAssetFileName";
+            experiment.logRemoteAsset(firstAssetLink, firstAssetFileName, false, metadata, context, onComplete);
+
+            awaitForCondition(onComplete, "first remote asset onComplete timeout", 30);
+
+            String secondAssetExpectedFileName = "secondAssetFile.extension";
+            URI secondAssetLink = new URI("s3://bucket/folder/" + secondAssetExpectedFileName);
+            experiment.logRemoteAsset(secondAssetLink, null, false, metadata, context, onComplete);
+
+            awaitForCondition(onComplete, "second remote asset onComplete timeout", 30);
+
+            // wait for assets become available and validate results
+            //
+            awaitForCondition(() -> experiment.getAssetList(ASSET_TYPE_ALL).size() == 2, "Assets was uploaded");
+            List<ExperimentAssetLink> assets = experiment.getAssetList(ASSET_TYPE_ALL);
+
+            validateRemoteAssetLink(assets, firstAssetLink, firstAssetFileName, metadataJson);
+            validateRemoteAssetLink(assets, secondAssetLink, secondAssetExpectedFileName, metadataJson);
```

Comment:
re-use the same metadataJson on both assert assertion. i would prefer first to be with metadata and second with no metadata (to eliminate the option that its passing by chance

another option is to have 2 different metadata for both assets 

I still think with and without is cleaner and tests more cases (the case with and without metadata)

Reply:
Fixed